h2/src/main/org/h2/mvstore/MVStore.java

```diff
@@ -1735,7 +1744,9 @@ public int compare(Chunk o1, Chunk o2) {
                 }
             }
             if (!queue.isEmpty()) {
-                result = queue;
+                ArrayList<Chunk> list = new ArrayList<>(queue);
```

Comment:
Suggest you using ArrayDeque here instead of ArrayList for iterator.remove() performance.